### PR TITLE
Introduces the option --overlay-type={subnet,full}, to be able to always generate IPIP tunnels regardless of node subnets

### PR DESCRIPTION
**Pull request by lucasmundim**
This PR implements PR #371 ideas, keeping the existing behaviour but allowing users to opt in for a full overlay regardless of the subnet the nodes are in

### CODE REVIEW

File: docs/user-guide.md [hunk 1]
```diff
@@ -49,9 +49,10 @@ Usage of kube-router:
...
+      --enable-overlay                   When enable-overlay is set to true, IP-in-IP tunneling is used for pod-to-pod networking across nodes in different subnets. When set to false no tunneling is used and routing infrastructure is expected to route traffic for pod-to-pod networking across nodes in different subnets (default true)
       --enable-pod-egress                SNAT traffic from Pods to destinations outside the cluster. (default true)
       --enable-pprof                     Enables pprof for debugging performance and memory leak issues.
+      --full-overlay                     When full-overlay is set to true, it changes "--enable-overlay=true" default behavior so that IP-in-IP tunneling is used for pod-to-pod networking across nodes regardless of the subnet the nodes are in. When set to false, the default, default "--enable-overlay=true" behavior is used
```
Comment (andrewsykim) [inline docs/user-guide.md]: What do you think about using `--overlay-type` instead, with a default value `regional` or `subnet`. And the other option is `type`?

Comment (lucasmundim) [reply docs/user-guide.md]: Sounds like a good idea to me. I don't know if I understood your question correctly ("And the other option is type?"). Could you elaborate a bit more?

Comment (andrewsykim) [reply docs/user-guide.md]: Sorry, typo. I meant "And the other option is full". 

So user needs to set both
```
--enable-overlay=true
--overlay-type={subnet,full}
```

Comment (andrewsykim) [reply docs/user-guide.md]: cc @murali-reddy

Comment (lucasmundim) [reply docs/user-guide.md]: > Sorry, typo. I meant "And the other option is full".

Now I got it :)

> So user needs to set both
> 
> ```
> --enable-overlay=true
> --overlay-type={subnet,full}
> ```

Yes, it makes sense. I've updated the code to reflect that.

**Comment by murali-reddy** [comment]
Did basic testing switching flag `--overlay-type={subnet,full}`. Code changes look good..

**Pull request merged**